src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,31 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animalName) {
+    if (animalName instanceof Herbivore && animalName.hidden === false) {
+      animalName.health -= 50;
+    }
+
+    if (animalName.health <= 0) {
```

Comment:
Ensure that `animalName` is an instance of `Animal` before accessing its `health` property. Consider adding a check to confirm `animalName` is an instance of `Animal` to avoid potential runtime errors.